src/pages/search.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect } from 'react'
+import { Link } from 'gatsby'
+import Layout from '../components/layout'
+import withSearch from '../components/utils/with-search'
+import searchPageStyle from './search.module.scss'
+import pressListStyle from '../components/common/press-list.module.scss'
+import { PublicationTitle } from '../components/common/publication'
+import DetailText from '../components/common/detail-text'
+import {
+  useSearch,
+  querySearch,
+  getHighlightResultOrExcerpt,
+} from '../context/search-context'
+
+export default withSearch(({ search }) => {
+  const [searchState, searchDispatch] = useSearch()
+  const { query, results } = searchState
+
+  function setQuery(value) {
+    return searchDispatch({ type: 'setQuery', payload: value })
+  }
+
+  useEffect(() => {
+    if (search.q) {
+      setQuery(search.q)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (query) {
+      querySearch(searchState, searchDispatch)
+    }
+  }, [query])
+
+  const totalHits =
+    (results.states.nbHits || 0) +
+    (results.blogPosts.nbHits || 0) +
+    (results.pages.nbHits || 0)
+  const hitsInfo = query.length
+    ? ` : ${query} (${totalHits} result${totalHits === 1 ? '' : 's'})`
+    : ''
+
+  let searchEvent
+
+  return (
+    <Layout title="Search">
+      <h2 className="hed-primary">Search{hitsInfo}</h2>
+      <form>
+        <label htmlFor="item">
+          Search
+          <input
+            type="text"
+            aria-label="Search"
+            id="item"
+            onChange={event => {
+              clearTimeout(searchEvent)
+              const { value } = event.currentTarget
+              searchEvent = setTimeout(() => {
+                setQuery(value)
+              }, 500)
+            }}
+          />
+        </label>
+        <button
+          type="submit"
+          onClick={async event => {
+            event.preventDefault()
+            await querySearch(searchState, searchDispatch)
+          }}
+        >
+          Search
+        </button>
+      </form>
+      {searchState.isFetching && (
+        <div>
+          <img
+            src="https://d1j8pt39hxlh3d.cloudfront.net/products/previews/RES3POBSZ353HFVPZOKR/2329_kKbUTLGEVXhuOIJqrT7QvIJFAXvEpQ3z.gif"
+            alt="Searching..."
+          />
+          <h3>Searching...</h3>
+        </div>
+      )}
+      <div className={searchPageStyle.searchResults}>
+        {/* State results */}
```

Comment:
These could probably be done DRY-er -- we could have a component for the results and then pass the inner content to the component for:
* state results
* blog posts
* pages

Plus, this should make it easier to add items like Twitter in the future!

Reply:
Totally agree. Once we've done the first review pass this ought to be a flexible component!